arangodb-net-standard/Transport/Http/HttpClientConfig.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+
+namespace ArangoDBNetStandard.Transport.Http
+{
+    public class HttpClientConfig
+    {
+        public Uri BaseAddress {
+            set {
+                _config += client => client.BaseAddress = value;
+                _key["baseaddress"] = value.ToString();
+            }
+        }
+        public MediaTypeHeaderValue ContentType { 
+            get {
+                return new MediaTypeHeaderValue(_contentTypeMap[_contentType]);
+            }
+        }
+
+        public void UseContentType(HttpContentType contentType)
+        {
+            _contentType = contentType;
+            _config += client =>
+            {
+                client.DefaultRequestHeaders.Accept.Clear();
+                client.DefaultRequestHeaders.Accept.Add(
+                    new MediaTypeWithQualityHeaderValue(_contentTypeMap[_contentType]));
+            };
+            _key["type"] = contentType.ToString();
+        }
+
+        private Dictionary<string, string> _key = new Dictionary<string, string>();
+        private Action<HttpClient> _config = _ => { };
+        private Action<HttpClient> _auth = _ => { };
+
+        private HttpClient Create() 
+        {
+            var cli = new HttpClient();
+            _config(cli);
+            _auth(cli);
+            return cli;
+        }
+
+        private HttpContentType _contentType;
+
+        private static readonly Dictionary<HttpContentType, string> _contentTypeMap =
+            new Dictionary<HttpContentType, string>
+            {
+                [HttpContentType.Json] = "application/json",
+                [HttpContentType.VPack] = "application/x-velocypack"
+            };
+
+        public void SetBasicAuth(string username, string password)
+        {
+            _auth = client =>
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(
+                    "Basic",
+                    Convert.ToBase64String(
+                        Encoding.ASCII.GetBytes($"{username}:{password}")));
+            _key["auth"] = $"basic,{username}:{password}";
+        }
+
+        public void SetJwtToken(string jwt)
+        {
+            _auth = client =>
+            {
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(
+                   "bearer",
+                   jwt);
+            };
+            _key["auth"] = $"jwt,{jwt}";
+        }
+
+        static ConcurrentDictionary<string, HttpClient> _pool = new ConcurrentDictionary<string, HttpClient>();
+
+        internal HttpClient Build()
+        {
+            return _pool.GetOrAdd(string.Join(";", _key), _ => Create());
```

Comment:
I'm not sure if this is sufficient. The `ConcurrentDictionary` documentation states:
> Since a key/value can be inserted by another thread while valueFactory is generating a value, you cannot trust that just because valueFactory executed, its produced value will be inserted into the dictionary and returned. If you call GetOrAdd simultaneously on different threads, valueFactory may be called multiple times, but only one key/value pair will be added to the dictionary.

So there is a chance that we create more `HttpClient` instances than necessary and I suppose these should be disposed right away. So I would suggest something like this:
```C#
  var key = string.Join(";", _key);
  HttpClient client;
  if (!_pool.TryGet(key, ref client) {
    var newClient = Create();
    client = _pool.GetOrAdd(key, newClient);
    if (client != newClient) {
      newClient.Dispose();
    }
  }
  return client;
  
```

Reply:
Discussed offline - since these freshly created `HttpClient` instances are not "open" there is no need to dispose them, so the current (simpler) solution is preferable.

Reply:
As we discussed, we should be aware that in rare occasions, there will be an extra HttpClient created and configured, but with no consequence (largely because of the way that HttpClient is used only for the Async part of its methods).